File: Chronir-Android/core/model/src/main/java/com/chronir/model/AlarmEnums.kt
package com.chronir.model

enum class TimezoneMode {
    FIXED,
    FLOATING
}

enum class PersistenceLevel {
    FULL,
    NOTIFICATION_ONLY,
    SILENT
}

enum class DismissMethod {
    SWIPE,
    HOLD_3S,
    SOLVE_MATH
}

enum class SyncStatus {
    LOCAL_ONLY,
    SYNCED,
    PENDING_SYNC,
    CONFLICT
}

enum class FollowUpInterval(val minutes: Int) {
    FIFTEEN_MINUTES(15),
    THIRTY_MINUTES(30),
    ONE_HOUR(60),
    TWO_HOURS(120),
    THREE_HOURS(180);

    val displayName: String
        get() = when (this) {
            FIFTEEN_MINUTES -> "15 min"
            THIRTY_MINUTES -> "30 min"
            ONE_HOUR -> "1 hour"
            TWO_HOURS -> "2 hours"
            THREE_HOURS -> "3 hours"
        }

    val timeIntervalMillis: Long
        get() = minutes * 60 * 1000L

    companion object {
        fun fromMinutes(minutes: Int): FollowUpInterval =
            entries.find { it.minutes == minutes } ?: THIRTY_MINUTES
    }
}

File: chronir/chronir/Core/Models/AlarmEnums.swift
import Foundation

enum CycleType: String, Codable, CaseIterable, Identifiable {
    case weekly
    case monthlyDate
    case monthlyRelative
    case annual
    case customDays
    case oneTime

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .weekly: return "Weekly"
        case .monthlyDate: return "Monthly"
        case .monthlyRelative: return "Monthly"
        case .annual: return "Annual"
        case .customDays: return "Custom"
        case .oneTime: return "One-Time"
        }
    }
}

enum TimezoneMode: String, Codable {
    case fixed
    case floating
}

enum PersistenceLevel: String, Codable {
    case full
    case notificationOnly
    case silent
}

enum DismissMethod: String, Codable {
    case swipe
    case hold3s
    case solveMath
}

enum SyncStatus: String, Codable {
    case localOnly
    case synced
    case pendingSync
    case conflict
}

enum PreAlarmOffset: String, Codable, CaseIterable, Identifiable, Comparable {
    case oneHour
    case oneDay
    case threeDays
    case sevenDays

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .oneHour: return "1 hour"
        case .oneDay: return "1 day"
        case .threeDays: return "3 days"
        case .sevenDays: return "7 days"
        }
    }

    var timeInterval: TimeInterval {
        switch self {
        case .oneHour: return 3600
        case .oneDay: return 86400
        case .threeDays: return 259200
        case .sevenDays: return 604800
        }
    }

    var notificationBody: String {
        switch self {
        case .oneHour: return "in 1 hour"
        case .oneDay: return "tomorrow"
        case .threeDays: return "in 3 days"
        case .sevenDays: return "in 1 week"
        }
    }

    /// Free tier only gets 1-day pre-alarm
    var requiresPlus: Bool {
        self != .oneDay
    }

    private var sortOrder: Int {
        switch self {
        case .oneHour: return 0
        case .oneDay: return 1
        case .threeDays: return 2
        case .sevenDays: return 3
        }
    }

    static func < (lhs: PreAlarmOffset, rhs: PreAlarmOffset) -> Bool {
        lhs.sortOrder < rhs.sortOrder
    }
}
